pokeapp/src/components/FilterBar.js

```diff
@@ -72,54 +93,136 @@ const [type, setType] = useState(false);
+      // If its not in the array add it to the array
+      setSelectedTypes((prevSelectedTypes) => [
+        ...prevSelectedTypes,
+        elementType,
+      ]);
+    } else {
+      // If its already in the array filter it out
+      setSelectedTypes((prevSelectedTypes) => {
+        return prevSelectedTypes.filter((el) => el !== elementType);
+      });
+    }
+  };
 
-  
+  const radioEl = (sortTypeEnum) => {
+    return (
+      <input
+        type="radio"
+        onClick={() => setSortType(sortTypeEnum)}
+        name="sortById"
+      />
+    );
+  };
+
+  const checkboxEl = (sortTypeEnum, type) => {
+    return (
+      <div
+        name={sortTypeEnum}
+        onClick={() => addRemoveTypeFilter(sortTypeEnum)}
+        className={selectedTypes.includes(sortTypeEnum) ? "active" : "inactive"}
+      >
+        {type}
+      </div>
+    );
+  };
 
   return (
     <div>
       <div>
         <label>Low to High</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.ID_LOW_TO_HIGH)} name="sortById" />
+        {radioEl(SortTypes.ID_LOW_TO_HIGH)}
+
         <label>High to Low</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.ID_HIGH_TO_LOW)} name="sortById" />
+        {radioEl(SortTypes.ID_HIGH_TO_LOW)}
+
         <label>A-Z</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.A_TO_Z)} name="sortAlphabetically" />
+        {radioEl(SortTypes.A_TO_Z)}
+
         <label>Z-A</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.Z_TO_A)} name="sortAlphabetically" />
+        {radioEl(SortTypes.Z_TO_A)}
+
         <label>Value High</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.VALUE_HIGH_TO_LOW)} name="sortByValue" />
+        {radioEl(SortTypes.VALUE_HIGH_TO_LOW)}
+
         <label>Value Low</label>
-        <input type="radio" onClick={()=>setSortType(SortTypes.VALUE_LOW_TO_HIGH)} name="sortByValue" />
+        {radioEl(SortTypes.VALUE_LOW_TO_HIGH)}
+      </div>
+
+      <div className="checkbox-type">
+        {checkboxEl(ElementTypes.FIRE, `Fire`)}
+        {checkboxEl(ElementTypes.WATER, `Water`)}
+        {checkboxEl(ElementTypes.ICE, `Ice`)}
+        {checkboxEl(ElementTypes.GRASS, `Grass`)}
+        {checkboxEl(ElementTypes.GHOST, `Ghost`)}
+        {checkboxEl(ElementTypes.GROUND, `Ground`)}
+        {checkboxEl(ElementTypes.FLYING, `Flying`)}
+        {checkboxEl(ElementTypes.BUG, `Bug`)}
+        {checkboxEl(ElementTypes.DARK, `Dark`)}
+        {checkboxEl(ElementTypes.DRAGON, `Dragon`)}
+        {checkboxEl(ElementTypes.ELECTRIC, `Electric`)}
+        {checkboxEl(ElementTypes.FAIRY, `Fairy`)}
+        {checkboxEl(ElementTypes.POISON, `Poison`)}
+        {checkboxEl(ElementTypes.FIGHTING, `Fighting`)}
+        {checkboxEl(ElementTypes.STEEL, `Steel`)}
+        {checkboxEl(ElementTypes.PSYCHIC, `Psychic`)}
+        {checkboxEl(ElementTypes.NORMAL, `Normal`)}
+        {checkboxEl(ElementTypes.ROCK, `Rock`)}
```

Comment:
You can use [Object.entries()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) to map over the enum, [then programmatically capitalize each first letter](https://stackoverflow.com/questions/1026069/how-do-i-make-the-first-letter-of-a-string-uppercase-in-javascript), or even better, have the enum contain the uppercased values and when you do your `pokemon.type.name` comparison, just use .toLowerCase().

Reply:
We will get on this on a new refactoring issue. Thanks for the idea!